Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt

```diff
@@ -62,79 +73,151 @@ class SurveyFragment : Fragment() {
-                                onDonePressed = { viewModel.computeResult(targetState) },
-                                onBackPressed = {
-                                    activity?.onBackPressedDispatcher?.onBackPressed()
+
+                        SurveyQuestionsScreen(
+                            surveyScreenData = surveyScreenData,
+                            isNextEnabled = isNextEnabled,
+                            onClosePressed = {
+                                shouldInterceptBackPresses = false
+                                activity?.onBackPressedDispatcher?.onBackPressed()
+                            },
+                            onPreviousPressed = { viewModel.onPreviousPressed() },
+                            onNextPressed = { viewModel.onNextPressed() },
+                            onDonePressed = { viewModel.onDonePressed() }
+                        ) { paddingValues ->
+
+                            val modifier = Modifier.padding(paddingValues)
+
+                            AnimatedContent(
+                                targetState = surveyScreenData,
+                                transitionSpec = {
+                                    val animationSpec: TweenSpec<IntOffset> =
+                                        tween(CONTENT_ANIMATION_DURATION)
+                                    val direction = getTransitionDirection(
+                                        initialIndex = initialState.questionIndex,
+                                        targetIndex = targetState.questionIndex,
+                                    )
+                                    slideIntoContainer(
+                                        towards = direction,
+                                        animationSpec = animationSpec,
+                                    ) with slideOutOfContainer(
+                                        towards = direction,
+                                        animationSpec = animationSpec
+                                    )
                                 }
-                            )
-                            is SurveyState.Result -> SurveyResultScreen(
-                                result = targetState,
-                                onDonePressed = {
-                                    activity?.onBackPressedDispatcher?.onBackPressed()
+                            ) { targetState ->
+
+                                when (targetState.surveyQuestion) {
+                                    SurveyQuestion.FREE_TIME -> {
+                                        FreeTimeQuestion(
+                                            modifier,
+                                            viewModel.freeTimeResponse,
+                                        ) { selected, answer ->
+                                            viewModel.onFreeTimeResponse(selected, answer)
+                                        }
```

Comment:
Simplify by passing a method reference:
```
FreeTimeQuestion(
  modifier,
  viewModel.freeTimeResponse,
  viewModel::onFreeTimeResponse
)
```

Same goes for the other question types.

Reply:
Good call! Updated!